analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotUseOutRefParameters.cs

```diff
@@ -79,4 +80,27 @@ public interface I3874
         void SetRef(ref I3874 obj); // Noncompliant
         void SetOut(out I3874 obj); // Noncompliant
     }
-}
+
+    public class Person
+    {
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+
+
+        public Person(string fname, string lname)
+        {
+            FirstName = fname;
+            LastName = lname;
+        }
+
+        public void Deconstruct(out string fname, out string lname) // Compliant,
+                                                                    // https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/deconstruct#user-defined-types
+        {
+            fname = FirstName;
+            lname = LastName;
+        }
+
+        public static void Deconstruct(out string foo) { foo = "foo"; } // Noncompliant
+        public static int Deconstruct(ref int bar) { return bar; } // Noncompliant
```

Comment:
We should add some tests also with extension methods as mentioned in the issue description.